engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/WorldSetupScreen.java

```diff
@@ -16,18 +16,213 @@
+
+        for (String label : params.keySet()) {
+
+            PropertyProvider provider = new PropertyProvider() {
+                @Override
+                protected <T> Binding<T> createTextBinding(Object target, FieldMetadata<Object, T> fieldMetadata) {
+                    return new WorldSetupScreen.WorldConfigBinding<>(worldConfig, label, compLib, fieldMetadata);
+                }
+
+                @Override
+                protected Binding<Float> createFloatBinding(Object target, FieldMetadata<Object, ?> fieldMetadata) {
+                    return new WorldSetupScreen.WorldConfigNumberBinding(worldConfig, label, compLib, fieldMetadata);
+                }
+            };
+
+            Component target = params.get(label);
+            List<Property<?, ?>> properties = provider.createProperties(target);
+            propLayout.addProperties(label, properties);
+        }
+    }
+
+    /**
+     * Updates a world configurator through setProperty() whenever Binding#set() is called.
+     */
+    private static class WorldConfigBinding<T> implements Binding<T> {
+        private final String label;
+        private final WorldConfigurator worldConfig;
+        private final FieldMetadata<Object, T> fieldMetadata;
+        private final ComponentLibrary compLib;
+
+        protected WorldConfigBinding(WorldConfigurator config, String label, ComponentLibrary compLib, FieldMetadata<Object, T> fieldMetadata) {
+            this.worldConfig = config;
+            this.label = label;
+            this.compLib = compLib;
+            this.fieldMetadata = fieldMetadata;
+        }
+
+        @Override
+        public T get() {
+            Component comp = worldConfig.getProperties().get(label);
+            return fieldMetadata.getValue(comp);
+        }
+
+        @Override
+        public void set(T value) {
+            T old = get();
+
+            if (!Objects.equals(old, value)) {
+                cloneAndSet(label, value);
+            }
+        }
+
+        private void cloneAndSet(String group, Object value) {
+            Component comp = worldConfig.getProperties().get(group);
+            Component clone = compLib.copy(comp);
+            fieldMetadata.setValue(clone, value);
+
+            // notify the world generator about the new component
+            worldConfig.setProperty(label, clone);
+        }
+    }
+
+    private static class WorldConfigNumberBinding implements Binding<Float> {
+
+        private WorldSetupScreen.WorldConfigBinding<? extends Number> binding;
+
+        @SuppressWarnings("unchecked")
+        protected WorldConfigNumberBinding(WorldConfigurator config, String label, ComponentLibrary compLib, FieldMetadata<Object, ?> field) {
+            Class<?> type = field.getType();
+            if (type == Integer.TYPE || type == Integer.class) {
+                this.binding = new WorldSetupScreen.WorldConfigBinding<>(config, label, compLib,
```

Comment:
You don't _really_ need to include a line break here, on a single line this is 138 chars, not too bad, our style warning is at 175 IIRC. Again somewhat personal/situational